sql/src/main/java/io/crate/operation/collect/sources/SystemCollectSource.java

```diff
@@ -132,4 +141,15 @@ public SystemCollectSource(ClusterService clusterService,
             new RowsCollector(downstream, toRowsIterable(collectPhase, iterableGetter.get(),
                 downstream.requirements().contains(Requirement.REPEAT))));
     }
+
+    /**
+     * Returns a new updater for a given table.
+     *
+     * @param ident the ident of the table
+     * @return a row updater instance for the given table
+     */
+    public SysRowUpdater<?> getRowUpdater(TableIdent ident){
```

Comment:
Would it make sense to de-couple this from the SystemCollectSource and have a SysUpdaterRegistry?

Or instead of depending on the whole SystemCollectSource only depend on `Function<TableIdent, SysRowUpdater<?>>` in ProjectionToProjectorVisitor